pymongo/uri_parser.py

```diff
@@ -62,21 +73,14 @@ def parse_userinfo(userinfo):
     """
     if '@' in userinfo or userinfo.count(':') > 1:
         raise InvalidURI("Username and password must be escaped according to "
-                         "RFC 3986, use urllib.parse.quote_plus")
-        # If we cannot round-trip a value, it is not properly URI encoded.
-        # This checks that we can take a value, quote it, and then unquote
-        # that quoted string and arrive at the same value.
-        # We take in the unquoted user and passwd strings and
-        # strip subdelimeters and then compare the original unquoted value
-        # without subdelimeters to the resulting string after round-tripping.
-        # One round trip means to quote once and then unquote.
-
-    unquoted_value_no_sdelims = "".join(
-        [ch for ch in userinfo if ch not in SUBDELIMS])
-    quoted_value = quote_plus(unquoted_value_no_sdelims)
-    if unquoted_value_no_sdelims != unquote_plus(quoted_value):
+                         "RFC 3986, use urllib.parse.quote")
+
+    no_subdelims = "".join(
+        [ch for ch in userinfo if ch not in _SUBDELIMS])
+
+    if unquoted_percent(no_subdelims):
```

Comment:
Do we need `_SUBDELIMS` at all here? I don't think it does anything for the unquoted_percent check now.

Reply:
Done.